src/EXTRA-FIX/fix_wall_flow.cpp

```diff
@@ -0,0 +1,294 @@
+
+#include "fix_wall_flow.h"
+#include "atom.h"
+#include "citeme.h"
+#include "comm.h"
+#include "domain.h"
+#include "error.h"
+#include "input.h"
+#include "lattice.h"
+#include "modify.h"
+#include "update.h"
+#include "variable.h"
+#include "random_mars.h"
+#include "memory.h"
+#include "force.h"
+#include "math_const.h"
+
+#include <cstring>
+#include <iostream>
+#include <algorithm>
+#include <functional>
+#include <fstream>
+
+using namespace LAMMPS_NS;
+using namespace FixConst;
+
+/* ---------------------------------------------------------------------- */
+
+static const char cite_fix_wall_flow_c[] =
+  "fix wall/flow command: doi:10.1177/10943420231213013\n\n"
+  "@Article{Pavlov-etal-IJHPCA-2024,\n"
+  " author = {Daniil Pavlov and Vladislav Galigerov and Daniil Kolotinskii and Vsevolod Nikolskiy and Vladimir Stegailov},\n"
+  " title = {GPU-based molecular dynamics of fluid flows: Reaching for turbulence},\n"
+  " journal = {The International Journal of High Performance Computing Applications},\n"
+  " year =    2024,\n"
+  " volume =  38,\n"
+  " number =  1,\n"
+  " pages =   34-49\n"
+  "}\n\n";
+
+FixWallFlow::FixWallFlow(LAMMPS *lmp, int narg, char **arg) :
+  Fix(lmp, narg, arg),
+  flowax(FlowAxis::AX_X),
+  flowvel(0.0),
+  flowdir(0),
+  rndseed(0),
+  current_segment(nullptr)
+{
+  if (lmp->citeme) lmp->citeme->add(cite_fix_wall_flow_c);
+  if (narg < 9) utils::missing_cmd_args(FLERR, "fix wall/flow", error);
+
+  dynamic_group_allow = 1;
+  bool do_abort = false;
+
+  int iarg = 3;
+  // parsing axis
+  if (strcmp(arg[iarg], "x") == 0) flowax = FlowAxis::AX_X;
+  else if (strcmp(arg[iarg],"y") == 0) flowax = FlowAxis::AX_Y;
+  else if (strcmp(arg[iarg],"z") == 0) flowax = FlowAxis::AX_Z;
+  else error->all(FLERR,"Illegal fix wall/flow argument: axis must by x or y or z, but {} specified", arg[iarg]);
+
+  ++iarg;
+  // parsing velocity
+  flowvel = utils::numeric(FLERR,arg[iarg],do_abort,lmp);
+  if (flowvel == 0.0) error->all(FLERR,"Illegal fix wall/flow argument: velocity cannot be 0");
+  if (flowvel > 0.0) flowdir = 1;
+  else flowdir = -1;
+  if(flowdir < 0) error->all(FLERR, "Illegal fix wall/flow argument: negative direction is not supported yet");
+
+  ++iarg;
+  // parsing temperature
+  double flowtemp = utils::numeric(FLERR,arg[iarg],do_abort,lmp);
+  kT = lmp->force->boltz * flowtemp / force->mvv2e;
+
+  ++iarg;
+  // parsing seed
+  rndseed = utils::inumeric(FLERR, arg[iarg],do_abort,lmp);
+  if(rndseed <= 0) error->all(FLERR, "Illegal fix wall/flow argument: random seed must be positive integer");
+
+  ++iarg;
+  // parsing wall count
+  int wallcount = utils::inumeric(FLERR,arg[iarg],do_abort,lmp);
+  if(wallcount <= 0) error->all(FLERR,"Illegal fix wall/flow argument: wall count must be positive integer");
+
+  ++iarg;
+  // parsing walls
+  if(narg - iarg != wallcount && narg - iarg != wallcount + 2) error->all(FLERR, "Wrong fix wall/flow wall count");
+  auto getscale = [&]() -> double {
```

Comment:
Why use a lambda function for such a trivial purpose? It just makes the code harder to read for people that are not C++ gurus.
Why not simply use simple if/else if/else if/else construct? The setting of scale in line 127 seems redundant.